alchemy/bin/services/execute-alchemy.ts

```diff
@@ -0,0 +1,148 @@
+import { log } from "@clack/prompts";
+import { execa } from "execa";
+import { resolve } from "node:path";
+import pc from "picocolors";
+import { zod as z } from "trpc-cli";
+import { exists } from "../../src/util/exists.ts";
+import { detectPackageManager } from "../services/package-manager.ts";
+
+export const entrypoint = z
+  .string()
+  .optional()
+  .describe(
+    "Path to the entrypoint file. Defaults to ./alchemy.run.ts > ./alchemy.run.js",
+  );
+
+export const watch = z
+  .boolean()
+  .optional()
+  .default(false)
+  .describe("Watch for changes to infrastructure and redeploy automatically");
+
+export const execArgs = {
+  cwd: z
+    .string()
+    .optional()
+    .describe("Path to the project directory (defaults to current directory)"),
+  quiet: z
+    .boolean()
+    .optional()
+    .default(false)
+    .describe("Suppress Create/Update/Delete messages"),
+  stage: z
+    .string()
+    .optional()
+    .describe(
+      "Specify which stage/environment to target. Defaults to your username ($USER, or $USERNAME on windows)",
+    ),
+  envFile: z.string().optional().describe("Path to environment file to load"),
+} as const;
+
+export async function execAlchemy(
+  main: string | undefined,
+  {
+    cwd = process.cwd(),
+    quiet,
+    stage,
+    destroy,
+    watch,
+    envFile,
+    read,
+    dev,
+  }: {
+    cwd?: string;
+    quiet?: boolean;
+    stage?: string;
+    destroy?: boolean;
+    watch?: boolean;
+    envFile?: string;
+    read?: boolean;
+    dev?: boolean;
+  },
+) {
+  const args: string[] = [];
+  const execArgs: string[] = [];
+  if (quiet) args.push("--quiet");
+  if (read) args.push("--read");
+  if (stage) args.push(`--stage ${stage}`);
+  if (destroy) args.push("--destroy");
+  if (watch) execArgs.push("--watch");
+  if (envFile) execArgs.push(`--env-file ${envFile}`);
+  if (dev) args.push("--dev");
+
+  // Check for alchemy.run.ts or alchemy.run.js (if not provided)
+  if (!main) {
+    const candidates = ["alchemy.run.ts", "alchemy.run.js"];
```

Comment:
Still may want to add an extra arg for this.

Reply:
That's what the main argument is for?

Reply:
For configuring which `alchemy.run` file to run. E.g. if you have a script in a different directory.

Reply:
Hmm, yes, you could do that. I think it's best to make it explicit though. What happens when you do that? Is a production deployment the default? Dev? Whatever you do, just make sure the intent is very clear.